csclient/csclient.go

```diff
@@ -0,0 +1,487 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// The csclient package provides access to the charm store API.
+package csclient
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"unicode"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v4"
+	"gopkg.in/macaroon-bakery.v0/httpbakery"
+
+	"github.com/juju/charmstore/params"
+)
+
+const (
+	apiVersion = "v4"
+
+	// ServerURL holds the default location of the global charm store.
+	// An alternate location can be configured by changing the URL
+	// field in the Params struct.
+	// For live testing or QAing the application, a different charm store
+	// location should be used, for instance
+	// "https://api.staging.jujucharms.com".
+	ServerURL = "https://api.jujucharms.com"
```

Comment:
I believe the canonical location should be https://api.jujucharms.com/charmstore